### DISCUSSION TITLE
Px now works on Linux!

### DISCUSSION THREAD

**Discussion by genotrance** [genotrance/px#136] [Announcements]
I'm very happy to announce that Px now works on Linux! Px [v0.6.0](https://github.com/genotrance/px/releases/tag/v0.6.0) is out and ready for test. There are some limitations though:
- Only NTLM and BASIC auth works
- PAC files are not supported yet
- Px only runs in a single process

None of these are fundamental limitations so it will just take some more iteration to make Linux on par with Windows. Of course, non-Windows systems will always need to configure the password since SSPI isn't available.

It is worth mention that v0.6.0 should be easy to get working on MacOSX, though I do not have an environment to test it out. It will be great if the community can provide patches to get it working.

Lastly, the changes for v0.6.0 were pretty drastic so I do expect some issues. I can only test Px in limited configurations so appreciate all issue reports in advance. Linux binaries have also been built for the first time for Ubuntu Focus and Alpine 3.15 using [Nuitka](https://nuitka.net) so I'll bet there will be issues there as well. Note that these are not static builds so portability is limited.

Looking forward to any issues or discussions here or on [Gitter](gitter.im/genotrance/px).

**Comment by eitsupi** [comment]
Thank you so much for working on this!
It is great that even on Linux there is the possibility of using px to work around some of the limitations that exist in CNTLM (auto-switching when out of proxy, etc.).

**Comment by genotrance** [comment]
Just posted v0.6.1 which added support for Px to run in multiple processes on Linux.

**Comment by genotrance** [comment]
[v0.8.0](https://github.com/genotrance/px/releases/tag/v0.8.0) that was just released adds PAC file support on Linux.

**Reply by genotrance** [reply]
Also a bit late but worth mention that Px v0.7.0 added support for NEGOTIATE on Linux thanks to the new libcurl backend.
